packages/@o3r/design/schematics/ extract-token/index.ts

```diff
@@ -0,0 +1,102 @@
+import type { Rule } from '@angular-devkit/schematics';
+import type { ExtractTokenSchematicsSchema } from './schema';
+import { posix, resolve } from 'node:path';
+import { AUTO_GENERATED_END, AUTO_GENERATED_START, DesignToken, DesignTokenGroup, DesignTokenNode, DesignTokenSpecification } from '../../src/public_api';
+
+const patternToDetect = 'o3r.var';
+
+/**
+ * Extract the token from o3r mixin sass file
+ * @param options
+ */
+export function extractToken(options: ExtractTokenSchematicsSchema): Rule {
+
+  const updateFileContent = (content: string): string => {
+    const start = content.indexOf(patternToDetect);
+    const end = content.lastIndexOf(patternToDetect);
+
+    if (start === -1 || end === -1 || !options.includeTags) {
```

Comment:
I would say the condition `end === -1` is redundant since it implies `start` would also be `-1`